docs/OSUpgrades.md

```diff
@@ -0,0 +1,74 @@
+# OS updates
+
+The MCO manages the Red Hat Enterprise Linux CoreOS operating system. Further,
+the operating system itself is just another part of the [release image](https://github.com/openshift/cluster-version-operator/), called `machine-os-content`.
+
+In other words, the cluster controls the operating system.
+
+# "Bootimage" vs machine-os-content
+
+We will use the term "bootimage" to mean an initial RHCOS disk image, such
+as an AMI, bare metal raw disk image, VMWare VMDK, OpenStack qcow2, etc.  
+These bootimages are built using [OSTree](https://github.com/ostreedev/ostree/)
+and [rpm-ostree](https://github.com/projectatomic/rpm-ostree/).
+
+Today, [the installer](https://github.com/openshift/installer/) pins the "bootimages"
+it uses, and released installers also pin the release image.  As noted above,
+release images contain `machine-os-content`, which can be a *different*
+RHCOS version.
+
+# The early pivot
+
+We do not want to require that a new bootimage is released for every update,
+and in general it can be hard to require that in every environment.  (For
+example, bare metal PXE setups)
+
+Hence, the MCO and installer combine to implement what is called "the early pivot".
+
+# Cluster bringup and upgrades
+
+Let's step through aspects of cluster bootstrap/installation and add some information
+about the early pivot.
+
+First, be sure you've read and understand [openshift/installer overview](https://github.com/openshift/installer/blob/37b99d8c9a3878bac7e8a94b6b0113fad6ffb77a/docs/user/overview.md).
+
+In this example we'll use AWS, but again this is all similar to a situation of
+booting bare metal machines via PXE or VMs in OpenStack.
+
+The openshift-installer starts, and uses the AMI it has pinned as the bootstrap node.
+
+The bootstrap node pulls the release image, which contains a reference to a newer `machine-os-content`.
```

Comment:
I've read this a few times, and I'm not clear on what exactly `machine-os-content` is. Is it a reference to some other image that holds an ostree repo? Or is it somehow a part of the release image? I've also read through the CVO documentation and did not find an answer there.

If I look at a release image, where/how would I find `machine-os-content`, and what would I see there?

Reply:
The release image is a reference to a lot of other images.  Look at e.g. this:

https://github.com/openshift/installer/blob/af56e2db2e2601156a1ebe56269c7deb67f3c725/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L23

As for what's *inside* it, see https://github.com/coreos/coreos-assembler/blob/a358e4475063c0f195508318c7f5c2a20af3b236/src/oscontainer.py

Reply:
> Is it a reference to some other image that holds an ostree repo?

Yes, exactly.